### PR TITLE
fix: rename unstable_redirect and add example

### PR DESCRIPTION
**Pull request by dai-shi**
https://github.com/dai-shi/waku/discussions/927#discussioncomment-10825901

### CODE REVIEW

File: examples/07_router/src/components/Counter.tsx [hunk 1]
```diff
@@ -13,7 +14,12 @@ export const Counter = () => {
       <button onClick={() => setCount((c) => c + 1)}>Increment</button>
       <h3>This is a client component.</h3>
       <span>path: {path}</span>
-      <Link to="/">Go to Home</Link>
+      <p>
+        <Link to="/">Go to Home</Link>
+      </p>
+      <p>
+        <button onClick={() => jump()}>Jump to random page</button>
```
Comment (rmarscher) [inline examples/07_router/src/components/Counter.tsx:21]: 😍 server functions are fun

**Review by rmarscher — APPROVED** [review]
LGTM. Thanks!

File: packages/waku/src/router/client.ts [hunk 2]
```diff
@@ -522,9 +522,13 @@ export function Router({ routerData = DEFAULT_ROUTER_DATA }) {
...
-                routerData[1]?.forEach((listener) =>
-                  listener(pathname, searchParamsString),
-                );
+                setTimeout(() => {
+                  // HACK we need to wait a tick for React to start rendering.
+                  // FIXME It doesn't feel ideal. Hope to find a better solution.
+                  routerData[1]?.forEach((listener) =>
+                    listener(pathname, searchParamsString),
+                  );
+                });
```
Comment (tylersayshi) [inline packages/waku/src/router/client.ts:531]: Would `startTransition` from react be less of a hack?

Comment (dai-shi) [reply packages/waku/src/router/client.ts]: Unfortunately, not. I'm not sure if it's React issue or Waku issue. Probably, the latter.

Meanwhile, I used another hack 3fedc84.

Comment (dai-shi) [reply packages/waku/src/router/client.ts]: Ahhh, unfortunately, this causes a flicker when moving to a new page. Let me revert it and open a new issue with `jump` case.

Comment (dai-shi) [reply packages/waku/src/router/client.ts]: #936

Comment (dai-shi) [reply packages/waku/src/router/client.ts]: #937

**Pull request merged**